ANES3-369 | upgrade to tailwind v4 and decanter v8

rebeccahongsf opened the pull request:
# READY FOR REVIEW

# Summary
- ANES3-369 | upgrade to tailwind v4 and decanter v8

# Review By (Date)
- Thursday, Oct 2

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out locally and do a visual comparison to the 1.x build.
2. Review code.

# Associated Issues and/or People
- ANES3-369

rebeccahongsf committed:
ANES3-369 | upgrade to tailwind v4 and decanter v8

rebeccahongsf committed:
add custom fonts

pookmish commented on package.json:
```diff
@@ -45,7 +45,7 @@
     "react-tiny-oembed": "^1.1.0",
     "sharp": "0.34.4",
     "tailwind-merge": "^2.6.0",
```
Update tailwind merge too.

pookmish commented on src/styles/index.css:
```diff
@@ -1,9 +1,211 @@
+  }
+  .data-\[intrail\=true\]\:block[data-intrail="true"] {
+    display: block;
+  }
```
This doesn't feel like the right way to do this. we want to add a selector for `data-intrail` that applies when the attribute is true. 
Looking at [this documenation](https://tailwindcss.com/docs/hover-focus-and-other-states#data-attributes), it looks like we just need to add something like:
`@custom-variant data-intrail (&[data-intrail~="true"]);`

pookmish commented on src/lib/gql/__generated__/drupal.d.tsx:
We don't want to modify any of the GQL files as part of this PR. If desired, lets do that as a separate PR.

rebeccahongsf committed:
cleanup

rebeccahongsf committed:
cleanup

rebeccahongsf committed:
update twmerge

rebeccahongsf committed:
rework aria and data styles

rebeccahongsf committed:
add bg gradients

rebeccahongsf committed:
add a white top gradient variant

pookmish commented on src/styles/index.css:
```diff
@@ -1,9 +1,226 @@
+  --font-merriweather: var(--font-merriweather);
+}
+
+@layer utilities {
```
So after digging into this a bit,  and this doesn't work the way it did in v3 TW. Using layers locks in the styles and they can't be used with variants such as responsive classes. ie we wouldn't be able to do `lg:grid-cols-1-1`. The only way to make it work with variants like that is to use the `@utility class-name {}` syntax. I even tried changing the layer to `@layer utilities` but it still doesn't work.

https://tailwindcss.com/docs/upgrade-guide#adding-custom-utilities

I think we need to change many of the layers into `@utility`. This might also be need to be done in Decanter.

pookmish reviewed (changes requested):
On top of the big issue, many styles on the site are very broken. The menu placement is a simple fix by adding `lg:top-100` so that's not much worry. But notice the widths, spacings, and the search input.

<img width="6626" height="2446" alt="image" src="https://github.com/user-attachments/assets/9c105e60-0da5-4561-9967-14c039fa089d" />

pookmish commented:
Also, Not sure if we need it, but the [documentation](https://tailwindcss.com/docs/installation/framework-guides/nextjs) says to add `@tailwindcss/postcss` dependency.

rebeccahongsf committed:
fixup utility styles and adjust masthead child component styles

rebeccahongsf commented:
> On top of the big issue, many styles on the site are very broken. The menu placement is a simple fix by adding `lg:top-100` so that's not much worry. But notice the widths, spacings, and the search input.
> 
> <img alt="image" width="2000" height="2000" src="https://private-user-images.githubusercontent.com/7185045/496241966-9c105e60-0da5-4561-9967-14c039fa089d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkzMzQ3NjksIm5iZiI6MTc1OTMzNDQ2OSwicGF0aCI6Ii83MTg1MDQ1LzQ5NjI0MTk2Ni05YzEwNWU2MC0wZGE1LTQ1NjEtOTk2Ny0xNGMwMzlmYTA4OWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDFUMTYwMTA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjIzNmJiNDlkNmNjNTNlYmY1OWQ4MTAzMGQ0YjZiYTFhODhlYWFhMmY5MWQzNjY3YTg3NTA1MTk0MzRjNTBhZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.b2IAQTFnTqKw3jf2574cHXTFWrBqzOM6cto4cxMC4jE">

Most of the styling is working as expected. The main difference comes from Decanter using an html root font size of `10px` (technically `62.5%`; see Decanter [base.css](https://github.com/SU-SWS/decanter/blob/v8/src/css/base/base.css#L21) and theme [font-size.css](https://github.com/SU-SWS/decanter/blob/v8/src/css/theme/font-size.css#L3) for `--text-root`), while Tailwind defaults to 16px. This accounts for the variations in widths, spacing, and padding.

For example:
- In Tailwind, `pb-2` equals `padding-bottom: 0.5rem` (5px).
- In Decanter, `pb-2 `equals `padding-bottom: 0.2rem` (2px).

I’m not sure it’s worth investing time in fixing the current masthead/main menu in the ANES repo, since their menu style is completely different. If we decide to reabsorb Decanter V8 into the `cardinalsites-nextjs` starter, we’ll need to do additional work to translate the existing Tailwind-specific styles into Decanter. 🤔 

As for fixing the main menu, I added `top-0` which should do the trick. I also updated the search bar and added `border max-w-300 w-full`.

rebeccahongsf commented:
> Also, Not sure if we need it, but the [documentation](https://tailwindcss.com/docs/installation/framework-guides/nextjs) says to add `@tailwindcss/postcss` dependency.

Great callout. I'm not sure if we need it as a dependency in this repo since it's been included as part of Decanter v8 but I'll do some testing this morning to see 😃

rebeccahongsf commented on src/styles/index.css:
```diff
@@ -1,9 +1,222 @@
+  list-style-type: upper-roman;
+}
+
+@layer components {
```
`@utility` should be working now. 

It seems like the `@layer components` aren't appearing though. I'm looking into the issue right now.

rebeccahongsf committed:
adjust styles and shift centered and gutters class from components to utility

cleanup new fonts

rebeccahongsf committed:
remove anes specific definitions
